pkg/cmd/pulumi/packagecmd/package_delete_test.go

```diff
@@ -0,0 +1,264 @@
+			packageVersion: "private/myorg/my-package@not-semver",
+			expectedErr:    "invalid semantic version",
+		},
+		{
+			name:           "registry backend error",
+			packageVersion: "private/myorg/my-package@1.0.0",
+			registryErr:    errors.New("failed to connect to registry"),
+			expectedErr:    "failed to get the registry backend",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var deletedSource, deletedPublisher, deletedName string
+			var deletedVersion semver.Version
+
+			mockCloudRegistry := &backend.MockCloudRegistry{
+				DeletePackageVersionF: func(
+					ctx context.Context, source, publisher, name string, version semver.Version,
+				) error {
+					deletedSource = source
+					deletedPublisher = publisher
+					deletedName = name
+					deletedVersion = version
+					return tt.deleteErr
+				},
+			}
+
+			testutil.MockBackendInstance(t, &backend.MockBackend{
+				GetCloudRegistryF: func() (backend.CloudRegistry, error) {
+					if tt.registryErr != nil {
+						return nil, tt.registryErr
+					}
+					return mockCloudRegistry, nil
+				},
+				GetReadOnlyCloudRegistryF: func() registry.Registry { return mockCloudRegistry },
+			})
+
+			cmd := newPackageDeleteCmd()
+			cmd.SetArgs([]string{tt.packageVersion, "--yes" /* Skip confirmation for tests */})
+			err := cmd.ExecuteContext(t.Context())
+
+			if tt.expectedErr != "" {
+				require.Error(t, err)
+				assert.Contains(t, err.Error(), tt.expectedErr)
+			} else {
+				require.NoError(t, err)
+				assert.Equal(t, tt.expectedDeleteArg.source, deletedSource)
+				assert.Equal(t, tt.expectedDeleteArg.publisher, deletedPublisher)
+				assert.Equal(t, tt.expectedDeleteArg.name, deletedName)
+				assert.Equal(t, tt.expectedDeleteArg.version, deletedVersion)
+			}
+		})
+	}
+}
+
+//nolint:paralleltest // This test uses the global backendInstance variable
+func TestPackageDeleteCmd_NonInteractiveRequiresYes(t *testing.T) {
+	mockCloudRegistry := &backend.MockCloudRegistry{
+		DeletePackageVersionF: func(
+			ctx context.Context, source, publisher, name string, version semver.Version,
+		) error {
+			t.Fatal("DeletePackageVersion should not be called without --yes in non-interactive mode")
+			return nil
+		},
+	}
+
+	testutil.MockBackendInstance(t, &backend.MockBackend{
+		GetCloudRegistryF: func() (backend.CloudRegistry, error) {
+			return mockCloudRegistry, nil
+		},
+		GetReadOnlyCloudRegistryF: func() registry.Registry { return mockCloudRegistry },
+	})
+
+	cmd := newPackageDeleteCmd()
+	cmd.SetArgs([]string{"private/myorg/my-package@1.0.0"})
+	err := cmd.ExecuteContext(t.Context())
+	// In non-interactive mode without --yes, should fail
+	// Note: This test assumes the test environment is non-interactive.
+	// The actual behavior depends on cmdutil.Interactive() which checks if stdin is a terminal.
+	if err != nil {
+		assert.Contains(t, err.Error(), "non-interactive mode requires --yes flag")
+	}
```

Comment:
This should be replaced with `assert.ErrorContains`
```suggestion
	assert.ErrorContains(t, err, "non-interactive mode requires --yes flag")
```

Otherwise the test succeeds even if there is no error.